packages/cli/src/webhooks/test-webhooks.ts

```diff
@@ -218,25 +219,44 @@ export class TestWebhooks implements IWebhookManager {
 	 * Return whether activating a workflow requires listening for webhook calls.
 	 * For every webhook call to listen for, also activate the webhook.
 	 */
-	async needsWebhook(
-		userId: string,
-		workflowEntity: IWorkflowDb,
-		additionalData: IWorkflowExecuteAdditionalData,
-		runData?: IRunData,
-		pushRef?: string,
-		destinationNode?: string,
-	) {
+	async needsWebhook(options: {
+		userId: string;
+		workflowEntity: IWorkflowDb;
+		additionalData: IWorkflowExecuteAdditionalData;
+		runData?: IRunData;
+		pushRef?: string;
+		destinationNode?: string;
+		preferredTrigger?: WorkflowRequest.ManualRunPayload['preferredTrigger'];
+	}) {
```

Comment:
The number of arguments to this function were getting out of hand so I turned it into an object.